taskboot/cargo.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import subprocess
+
+from taskboot.config import Configuration
+
+logger = logging.getLogger(__name__)
+
+
+def cargo_publish(target, args):
+    """
+    Publish a crate on crates.io
+    """
+
+    # Load config from file/secret
+    config = Configuration(args)
+    assert config.has_cargo_auth(), "Missing Cargo authentication"
+
+    # Build the package to publish on crates.io
+    try:
+        subprocess.run(["cargo", "publish", "--dry-run"], check=True)
+    except subprocess.CalledProcessError:
+        raise Exception("Failed to build the package to publish on crates.io")
```

Comment:
Nit: you can just rely on the exception thrown by subprocess.run, no need to catch and throw a custom one:
```suggestion
subprocess.run(["cargo", "publish", "--dry-run"], check=True)
```